services/frontend-service/src/ui/components/SideBar/SideBar.tsx

```diff
@@ -176,11 +178,69 @@ export const SideBarListItem: React.FC<{ children: BatchAction }> = ({ children:
     const actionDetails = getActionDetails(action, appLocks, environmentLocks);
 
     const handleDelete = useCallback(() => deleteAction(action), [action]);
+    const similarLocks = useLocksSimilarTo(action);
+    const handleAddAll = useCallback(() => {
+        similarLocks.appLocks.forEach((displayLock: DisplayLock) => {
+            if (!displayLock.environment) {
+                throw new Error('action of type ' + ActionTypes.DeleteEnvironmentLock + ' must have environment set');
```

Comment:
Is the Action type here correct?
If the environment is not optional do we need to check this here too?

Reply:
the errors are outdated, I will update them